### PR TITLE
Fix panic caused by invalid spanId with b3 propagator

### PR DESCRIPTION
**Pull request by Cirilla-zmh**
### Background
When using the b3 single propagator, the presence of an invalid spanId in the request can trigger a panic.

### Cause
The panic occurs due to an array index out-of-bounds error when parsing the `SpanID` without proper length checks for invalid spanIds.

### Fix
Added a length check for `SpanID` to prevent this panic by ensuring spanIds are of valid length before processing.

### CODE REVIEW

File: propagators/b3/b3_propagator_test.go [hunk 1]
```diff
@@ -217,6 +217,16 @@ func TestExtractSingle(t *testing.T) {
 		{"000000000000007b", trace.SpanContextConfig{}, errInvalidScope, false, false},
 		{"000000000000007b00000000000001c8", trace.SpanContextConfig{}, errInvalidScope, false, false},
 		// Support short trace IDs.
```
Comment (dmathieu) [inline propagators/b3/b3_propagator_test.go:219]: This comment seems to apply to the other test cases, not the two you just added.
There should probably be a quick explanation for those two cases, such as `Handle invalid Trace and Span IDs`.

File: CHANGELOG.md [hunk 2]
```diff
@@ -18,6 +18,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - The superfluous `response.WriteHeader` call in `go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp` when the response writer is flushed. (#5634)
 - Custom attributes targeting metrics recorded by the `go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp` are not ignored anymore. (#5129)
 - Use `c.FullPath()` method to set `http.route` attribute in `go.opentelemetry.io/contrib/instrumentation/github.com/gin-gonic/gin/otelgin`. (#5734)
+- Fix out-of-bounds panic caused by invalid spanId with b3 propagator. (#5754)
```
Comment (dmathieu) [inline CHANGELOG.md:21]:
```suggestion
- Fix out-of-bounds panic in case of invalid span ID in `go.opentelemetry.io/contrib/propagators/b3`. (#5754)
```

**Review by MadVikingGod — APPROVED** [review]

File: propagators/b3/b3_propagator_test.go (hunk 1 above)
Comment (Cirilla-zmh) [reply propagators/b3/b3_propagator_test.go]: Sorry, my bad. I have resolved them.

**Comment by dmathieu** [comment]
The changelog entry needs to be fixed.

**Comment by Cirilla-zmh** [comment]
> The changelog entry needs to be fixed.

@dmathieu Thank you for your hint. I just resolved the conflict and need your approval to run the workflow.

**Review by dmathieu — APPROVED** [review]

**Comment by Cirilla-zmh** [comment]
@dmathieu @MadVikingGod 
I've just rebased the main branch. If it's convenient for you, could you please run the workflow again? Thank you!

**Comment by Cirilla-zmh** [comment]
@dmathieu @MadVikingGod 
It seems that this PR already has the conditions for merge. If convenient, please merge it. Thanks a lot for you!

**Review by hanyuancheung — APPROVED** [review]

**Review by MrAlias — APPROVED** [review]

**Pull request merged**